metaflow/plugins/aws/step_functions/step_functions.py

```diff
@@ -307,19 +310,20 @@ def _visit(node, workflow, exit_node=None):
                 workflow.add_state(state.next(cardinality_state_name))
                 cardinality_state = (
                     State(cardinality_state_name)
-                    .dynamo_db(SFN_DYNAMO_DB_TABLE, "$.JobId", "for_each_cardinality")
+                    .dynamo_db(SFN_DYNAMO_DB_TABLE, "$.JobId", "for_each_cardinality, root_run_id")
                     .result_path("$.Result")
                 )
                 iterator_name = "*%s" % node.out_funcs[0]
                 workflow.add_state(cardinality_state.next(iterator_name))
                 workflow.add_state(
-                    Map(iterator_name)
+                    Map(iterator_name, self.use_distributed_map)
                     .items_path("$.Result.Item.for_each_cardinality.NS")
                     .parameter("JobId.$", "$.JobId")
                     .parameter("SplitParentTaskId.$", "$.JobId")
                     .parameter("Parameters.$", "$.Parameters")
                     .parameter("Index.$", "$$.Map.Item.Value")
-                    .next(node.matching_join)
+                    .parameter("RootRunId.$", "$.Result.Item.root_run_id.S")
```

Comment:
Here we are extracting the root Run ID from dynamo and putting into the parameters as `RootRunId` so it can be used.